### PR TITLE
Static analyzer (PVS Studio) fixes

### PR DESCRIPTION
**Pull request by dimhotepus**
Some fixes from PVS Studio static analyzer.

Product page: https://www.viva64.com/en/pvs-studio/
How to use for free: https://www.viva64.com/en/b/0600/

Thanks for SDK, used that in own projects.

### CODE REVIEW

**Comment by dimhotepus** [comment]
Build is broken due to artifacts to upload limit and name resolution failure.
```
failed: Temporary failure in name resolution.
wget: unable to resolve host address ‘www.openssl.org’
make: *** [openssl-1.1.0j.tar.gz] Error 4
Makefile:110: recipe for target 'openssl-1.1.0j.tar.gz' failed

##[error]Bash exited with code '1'.
```

File: Release/src/streams/fileio_win32.cpp [hunk 1]
```diff
@@ -496,7 +496,7 @@ size_t _read_file_async(_In_ streams::details::_file_info_impl* fInfo,
 
     size_t result = static_cast<size_t>(-1);
 
-    if (wrResult == TRUE)
+    if (wrResult != FALSE)
```
Comment (BillyONeal) [inline Release/src/streams/fileio_win32.cpp]: I think a better fix for these would be to just remove the comparison; `if (wrResult)`

File: Release/src/utilities/asyncrt_utils.cpp [hunk 2]
```diff
@@ -1148,7 +1148,7 @@ datetime __cdecl datetime::from_string(const utility::string_t& dateString, date
             {
                 tzHours = 8;
             }
-            else if ((tzCh == _XPLATSTR('+') || tzCh == _XPLATSTR('-')) && ascii_isdigit2(str[1]) &&
+            else if ((str[0] == _XPLATSTR('+') || str[0] == _XPLATSTR('-')) && ascii_isdigit2(str[1]) &&
```
Comment (BillyONeal) [inline Release/src/utilities/asyncrt_utils.cpp]: Hmmm this suggests a missing test?

Comment (BillyONeal) [reply Release/src/utilities/asyncrt_utils.cpp]: Oh I see you added it. Thanks!

**Pull request merged**